dist/bin/scaladoc

```diff
@@ -127,7 +127,6 @@ eval "\"$JAVACMD\"" \
      ${JAVA_OPTS:-$default_java_opts} \
      "${java_args[@]}" \
      "${jvm_cp_args-}" \
-     -Dscala.usejavacp=true \
```

Comment:
Any particular reason for removing this flag? It's in dist/bin/scala and dist/bin/scalac, so it makes sense to have it here too.

Reply:
Yes, I removed this flag intentionally. It helped me a lot and made most of things working but then I realized that with this flag set, generating standard library documentation is failing. I'm not sure about the exact reason but I believe it's because we've got then two standard libraries on classpath and there was some conflict between them. Instead of setting the flag there, I set it directly in tasks that needed it.

Reply:
Ok, but that means the scaladoc binary isn't really usable for external users. I believe the best solution is to not call the scaladoc script from Build.scala at all. Instead you should be able to run the scaladoc main class from the sbt project directly: `(Compile / runMain).toTask("-foo -bla -bar")`

Reply:
**Caution !** scaladoc binary MUST BE usable for external users, as are `javadoc` and `haddock` for instance.

Reply:
So probably I need to implement @smarter  solution. The funny part is that the -usejavacp option is not there on master, I added it some commits before in this PR, but now I realized it breaks doc generation for one task.